consensus/src/consensus.rs

```diff
@@ -201,29 +200,25 @@ impl<T: Operations + 'static, D: Database + 'static> Consensus<T, D> {
             // Consensus loop
             // Initialize and run consensus loop
 
-            let mut iteration_counter: u8 = 0;
+            let mut iter: u8 = 0;
             let mut iter_ctx = IterationCtx::new(
                 ru.round,
-                iteration_counter,
-                proposal_handler.clone(),
-                validation_handler.clone(),
-                ratification_handler.clone(),
+                iter,
+                proposal_handler,
+                validation_handler,
+                ratification_handler,
             );
 
-            loop {
-                iter_ctx.on_begin(iteration_counter);
+            while iter < CONSENSUS_MAX_ITER {
+                iter_ctx.on_begin(iter);
 
                 let mut msg = Message::empty();
                 // Execute a single iteration
-                for pos in 0..phases.len() {
-                    let phase = phases.get_mut(pos).unwrap();
-
-                    let step = iteration_counter.step_from_pos(pos);
-                    let name = phase.name();
-
+                for phase in phases.iter_mut() {
+                    let step_name = phase.to_step_name();
                     // Initialize new phase with message returned by previous
                     // phase.
-                    phase.reinitialize(&msg, ru.round, step).await;
+                    phase.reinitialize(&msg, ru.round, iter).await;
```

Comment:
Not sure it makes sense to call this `reinitialize` anymore. It should probably be `initialize`

Reply:
Can we agree that is not on the scope of this PR?
No problem in change it, but I would like to first agree on what the name should be (if)

Reply:
Sure, it's not part of this PR, that's why I just added a comment.
I think `initialize` is fine, but if you have something more aligned to standard (or our) naming convention, go for it